### ISSUE TITLE
How to handle multi-modal images?

### ISSUE THREAD

**Issue by meghbhalerao** [fepegar/torchio#236]
Hi!
I am training a model on multi-modal data and defining my individual subject something like this:

```
one_subject = Subject(
        channel_1=Image('../BRATS2018_crop_renamed/LGG75_T1.nii.gz', torchio.INTENSITY),
        channel_2=Image('../BRATS2018_crop_renamed/LGG75_T2.nii.gz', torchio.INTENSITY),
        channel_3=Image('../BRATS2018_crop_renamed/LGG75_T3.nii.gz', torchio.INTENSITY) # this is not a real modality, but just for example,
        label=Image('../BRATS2018_crop_renamed/LGG75_Label.nii.gz', torchio.LABEL),
    )
```
I am making a list of multiple such subjects. After which I create a data loader object from this  using:

```
subjects_dataset = ImagesDataset(subjects)
batch_loader = DataLoader(
        subjects_dataset,
        batch_size=batch_size,
        )
```

Now my question is: when I iterate through this dataloader, how do I access the 4D image (which in this case would be channel * H * L * W - i.e. one composed of all the three modalities), from the iterable - since a 4D image is what my 3D segmentation model expects? Basically, it seems like the Subject class does not concatenate the individual modalities (TorchIO.INTENSITY) by default? Please let me know if something is not clear or if I am missing something.
Thank you!

**Comment by fepegar** [comment]
Hi, @meghbhalerao. Have you tried running that?

The `batch_loader` will return a dictionary with keys `channel_1`, `channel_2`, `channel_3` and `label`, not a 4D image.

Maybe some code will help:

```python
In [1]: import torch
imo
In [2]: import torchio as tio

In [3]: one_subject = tio.Subject(
   ...:     channel_1=tio.ScalarImage('colin27_t1_tal_hires.nii'),
   ...:     channel_2=tio.ScalarImage('colin27_t2_tal_hires.nii'),
   ...:     channel_3=tio.ScalarImage('colin27_pd_tal_hires.nii'),
   ...:     label=tio.LabelMap('colin27_cls_tal_hires.nii'),
   ...: )

In [4]: subjects_dataset = tio.ImagesDataset([one_subject])

In [5]: batch_loader = torch.utils.data.DataLoader(subjects_dataset)

In [6]: batch = next(iter(batch_loader))

In [7]: batch.keys()
Out[7]: dict_keys(['channel_1', 'channel_2', 'channel_3', 'label'])

In [8]: batch['channel_1'][tio.DATA].shape
Out[8]: torch.Size([1, 1, 362, 434, 362])
```

**Comment by fepegar** [comment]
Then, if you want, you can concatenate your images. For example:

```python
In [16]: all_images = torch.cat([batch[key][tio.DATA] for key in batch.keys()], dim=1)

In [17]: all_images.shape
Out[17]: torch.Size([1, 4, 362, 434, 362])
```

Would that work for you?

**Comment by meghbhalerao** [comment]
Hi @fepegar!
Thanks a lot for your prompt reply! Yes, then I would have to concatenate the images during the iteration (runtime). Thanks again!
